rpc/dialer.go

```diff
@@ -335,8 +331,17 @@ func dialDirectGRPC(ctx context.Context, address string, dOpts dialOptions, logg
 			connPtr = &rpcCreds.conn
 		}
 		dialOpts = append(dialOpts, grpc.WithPerRPCCredentials(rpcCreds))
+		unaryInterceptors = append(unaryInterceptors, UnaryClientReauthInterceptor(rpcCreds))
+		// ReauthInterceptors will not work for streaming calls; we can only
+		// intercept the creation of a stream, and the ensuring of authentication
+		// server-side happens per RPC request (per usage of the stream).
```

Comment:
Discussed this with @cheukt offline; I can't find a great way to clear the `accessToken` _per usage_ of the gRPC stream. This means that if `streamClient.RecvMsg` returns an unauthenticated error, all subsequent calls to `RecvMsg` will return the same error.